system-addon/lib/ThemeFeed.jsm

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+
+const {actionCreators: ac, actionTypes: at} = ChromeUtils.import("resource://activity-stream/common/Actions.jsm", {});
+
+const THEME_UPDATE_EVENT = "lightweight-theme-styling-update";
+
+this.ThemeFeed = class ThemeFeed {
+  init(LightweightThemeManager) {
+    Services.obs.addObserver(this, THEME_UPDATE_EVENT);
+
+    let temp = {LightweightThemeManager};
+    ChromeUtils.import("resource://gre/modules/LightweightThemeManager.jsm", temp);
+    this.updateTheme(temp.LightweightThemeManager.currentThemeForDisplay);
+  }
+
+  uninit() {
+    Services.obs.removeObserver(this, THEME_UPDATE_EVENT);
+  }
+
+  observe(subject, topic, data) {
+    if (topic === THEME_UPDATE_EVENT) {
+      this.updateTheme(JSON.parse(data));
+    }
+  }
+
+  _parseRGB(colorString) {
+    let rgb = colorString.match(/^rgba?\((\d+), (\d+), (\d+)/);
+    rgb.shift();
+    return rgb.map(x => parseInt(x, 10));
+  }
+
+  updateTheme(data) {
+    if (data && data.window) {
+      // We only update newtab theme if the theme activated isn't window specific.
+      // We'll be able to do better in the future: see Bug 1444459
```

Comment:
Fun fact, different windows can be assigned different themes via web extensions, for example. Also private browsing. Out of scope for now :)